### DISCUSSION TITLE
Serious problems at registering with X-Plane

### DISCUSSION THREAD

**Discussion by Glasairman** [curiosity-workshop/XPlane-Direct-superceded#7] [Q&A]
I have a 5% success rate in registering 3 (trim) variables, I have tried moving the Requests within the loop() to no avail . DataRefs are correct.

 Xinterface.begin("Xplane Demo register");  
  delay(200);
  Xinterface.registerDataRef("sim/flightmodel/controls/elv_trim", XPL_READ, 500, 10, &state1,0);
  Xinterface.registerDataRef("sim/flightmodel/controls/ail_trim", XPL_READ, 500, 10, &state2,1);
  Xinterface.registerDataRef("sim/flightmodel/controls/rud_trim", XPL_READ, 500, 10, &state3,2); 

also tried  restarting the plugin, and XPLDirect registration, totally frustrating

It would be useful to have a return code from Xinterface calls. Full code available if required

**Comment by curiosity-workshop** [comment]
Send the code ill try it here.  Which board are you using?

Also send xpldirecterror.log if possible

**Comment by Glasairman** [comment]
Hi, thanks for your offer of help. 

Note that the use of the XPLDirect calls inside the loop was temporary, and didn't make any difference. . "Status" from the Plugin shows no registration or failure to register with possibly wrong datarefs (all were correct) or timeouts
 
Mit freundlichen Grüssen / Best Regards

David Walker
Jahnstrasse 64
D-82538 GeretsriedGermany

Tel.08171.2385010
Mob. 0176.55225959
 
 
-----Ursprüngliche Mitteilung-----
Von: curiosity-workshop ***@***.***>
An: curiosity-workshop/XPlane-Direct ***@***.***>
Cc: Glasairman ***@***.***>; Author ***@***.***>
Verschickt: Fr, 9. Apr. 2021 15:28
Betreff: Re: [curiosity-workshop/XPlane-Direct] Serious problems at registering with X-Plane (#7)

Send the code ill try it here. Which board are you using?

Also send xpldirecterror.log if possible

**Comment by Glasairman** [comment]
Sorry Board is a Uno Clone with CH340 driver
 
Mit freundlichen Grüssen / Best Regards

David Walker
Jahnstrasse 64
D-82538 GeretsriedGermany

Tel.08171.2385010
Mob. 0176.55225959
 
 
-----Ursprüngliche Mitteilung-----
Von: curiosity-workshop ***@***.***>
An: curiosity-workshop/XPlane-Direct ***@***.***>
Cc: Glasairman ***@***.***>; Author ***@***.***>
Verschickt: Fr, 9. Apr. 2021 15:28
Betreff: Re: [curiosity-workshop/XPlane-Direct] Serious problems at registering with X-Plane (#7)

Send the code ill try it here. Which board are you using?

Also send xpldirecterror.log if possible

**Reply by curiosity-workshop** [reply]
I don't see an attachment.  If you are having trouble email me directly flightpro42@gmail.com

**Reply by curiosity-workshop** [reply]
One thing I noticed just glancing at it is that you were using the array overloads for the registrations on datarefs that aren't array types.

I modified what you have and compiled, all registered fine like this:

 Xinterface.registerDataRef("sim/flightmodel/controls/elv_trim", XPL_READ, 500, 10, &state1);
 Xinterface.registerDataRef("sim/flightmodel/controls/ail_trim", XPL_READ, 500, 10, &state2);
 Xinterface.registerDataRef("sim/flightmodel/controls/rud_trim", XPL_READ, 500, 10, &state3);
